packager/packager.cc

```diff
@@ -1003,6 +1009,10 @@ struct PackageInfoFinder {
             send.addPosArg(prependName(move(importArg), core::Names::Constants::PackageSpecRegistry()));
         }
 
+        if (send.fun == core::Names::exportAll() && send.numPosArgs() == 0) {
+            info->exportAll_ = true;
+        }
```

Comment:
I don't know whether we have tests for other package spec methods like this, but could we add a test passing arguments to `export_all!` and verify that errors are detected?

Reply:
Yup! I've added that. The way typechecking these works is that we actually treat them as normal Ruby methods, so they're subject to the same typechecking constraints as any other methods. (In particular, when we enter these methods into `GlobalState` (e.g. [here](https://github.com/sorbet/sorbet/pull/6647/files#diff-2c71c670c6721566158753a99c725a4826efc01a8a6a3c53a32c98fe3889252c)) we also give their argument structure; this one takes no parameters, so if you pass it a parameter, then inference will raise the same error it would raise for any method with too many arguments.)